### PR TITLE
Generate shape-independent graph if any input dimension < 2

### PR DESCRIPTION
**Pull request by wschin**

### CODE REVIEW

File: orttraining/orttraining/core/graph/gradient_builder.cc [hunk 1]
```diff
@@ -194,7 +194,8 @@ IMPLEMENT_GRADIENT_BUILDER(GetMatMulGradient) {
     }
   };
 
-  if (A_has_shape && B_has_shape && Y_has_shape) {
+  if (A_has_shape && B_has_shape && Y_has_shape &&
+      A_shape.size() >= 2 && B_shape.size() >= 2) {
```
Comment (SherlockNoMad) [inline orttraining/orttraining/core/graph/gradient_builder.cc:198]: Matmul schema only allow inputs with dim>=2. 
I thought this is an implicit prerequisite? Are you seeing inputs with less then 2 dim?

**Review by SherlockNoMad — APPROVED** [review]

File: orttraining/orttraining/core/graph/gradient_builder.cc (hunk 1 above)
Comment (SherlockNoMad) [reply orttraining/orttraining/core/graph/gradient_builder.cc:198]: I saw the discussion on team's channel, and understand now this is a torch bug/"feature". 
This fix is harmless.

Comment (yuslepukhin) [reply orttraining/orttraining/core/graph/gradient_builder.cc:198]: Out MatMul allows vectors.

---
In reply to: [571201490](https://github.com/microsoft/onnxruntime/pull/6581#discussion_r571201490) [](ancestors = 571201490)

**Pull request merged**